### PR TITLE
Superseded Do Not Merge Implement aws guardduty

### PR DESCRIPTION
**Pull request by pwyborn**
Why
ministryofjustice/cloud-platform#532
Initial go at AWS Guarduty (main account Moj-cp only)
Would appreciate your input please
It does seem to be functioning in cp account and pagerduty

### CODE REVIEW

**Comment by pwyborn** [comment]
Hi @jasonBirchall  - would it be possible to review please. I think the only thing I have not done now is a runbook (might need to check with you what is required here). I now have the cp account as master with the moj-pi account as member.
I have disabled the pagerduty service for GuardDuty at present - as a port is being being probed on moj-pi (low priority alert - need to think of the action to be taken in these cases)?

**Comment by pwyborn** [comment]
@jasonBirchall - Terraform now has multiple accounts provision

Have also made it so that only medium/high severity findings alert (I could make it just high)

Moj-dsd and moj-intra will fail a "terraform apply" at the moment - as someone has already set up guardduty there (probably as a test). 
I would need to delete the guardduty there and then rerun "terraform apply". An invitation would then be sent out to the root email.

File: terraform/global-resources/aws_guardduty/README.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+Also see here [Aws Cloudwatch Frequently Asked Questions](https://aws.amazon.com/guardduty/faqs/ where most general information regarding this service can be found)
+
+Consideration will have to be given as to whether GuardDuty is set up here in [moj-cloud-platform](https://moj-cloud-platform-test-2.eu.auth0.com/samlp/WAgw4FygIHs1Vny6whAjfnem6BiUr4qv) or whether it should be in the digital AWS route account. You will then be able to set up control of AWS subaccounts from here.
+
+Please also see [terraform aws GuardDuty detector ](https://www.terraform.io/docs/providers/aws/r/guardduty_detector.html) regarding setting up the config (here) for terraform GuardDuty detector
+
+## Table of contents
+  - [1 Prerequisite files](#1-prerequisite-files)
+  - [2 Pagerduty to Slack integration](#2-pagerduty-to-slack-integration)
```
Comment (jasonBirchall) [inline terraform/global-resources/aws_guardduty/README.md]: This link doesn't work at the moment. I think it's just the naming.

File: terraform/global-resources/aws_guardduty/event-pattern.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        8.3,
+        8.4,
+        8.5,
+        8.6,
+        8.7,
+        8.8,
+        8.9
+      ]
+    }
+  }
```
Comment (jasonBirchall) [inline terraform/global-resources/aws_guardduty/event-pattern.json]: As discussed, create a dir here `/terraform/global-resources/resources` and store additional files.

File: terraform/global-resources/aws_guardduty/iplist.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+156.67.33.245\32
```
Comment (jasonBirchall) [inline terraform/global-resources/aws_guardduty/iplist.txt]: Same here. Place in `resources` dir.

File: terraform/global-resources/aws_guardduty/main.tf [hunk 4]
```diff
@@ -0,0 +1,427 @@
...
+# membership7 account GuardDuty member
+# -----------------------------------------------------------
+
+resource "aws_guardduty_member" "member7" {
+  account_id         = "${aws_guardduty_detector.member7.account_id}"
+  detector_id        = "${aws_guardduty_detector.master.id}"
+  email              = "${var.member7_email}"
+  invite             = true
+  invitation_message = "please accept guardduty invitation"
+}
```
Comment (jasonBirchall) [inline terraform/global-resources/aws_guardduty/main.tf]: I recommend renaming this file and running it in the same state as `/terraform/global-resources`

File: terraform/global-resources/aws_guardduty/variables.tf [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  default = "guardduty-admin"
+}
+
+variable "tags" {
+  default = {
+    "owner"   = "moj-platform"
+    "project" = "cp-guardduty"
+    "client"  = "Internal"
+  }
+}
```
Comment (jasonBirchall) [inline terraform/global-resources/aws_guardduty/variables.tf]: Use the same variables file as `/terraform/global-resources/variables.tf`

**Comment by pwyborn** [comment]
Superceded by https://github.com/ministryofjustice/cloud-platform-infrastructure/pull/141

**Comment by pwyborn** [comment]
Superseded by https://github.com/ministryofjustice/cloud-platform-infrastructure/pull/117

**Comment by pwyborn** [comment]
Superseded by https://github.com/ministryofjustice/cloud-platform-infrastructure/pull/141

**Pull request closed without merging**